src/Widgets/HeaderBar.vala

```diff
@@ -92,20 +76,19 @@ public class Akira.Widgets.HeaderBar : Gtk.Box {
         toolset = new Akira.Partials.MenuButton ("insert-object", _("Add"), _("Add a New Object"));
         toolset.popup = tools;
 
-        toolbar.add (toolset);
-        toolbar.add (new Akira.Partials.Spacer (35));
-
         settings = new Akira.Partials.MenuButton ("open-menu", _("Settings"), _("Open Settings"));
 
         layout = new Akira.Partials.HeaderBarButton ("preferences-system-windows", _("Layout"), _("Toggle Layout (Ctrl+.)"));
         ruler = new Akira.Partials.HeaderBarButton ("applications-accessories", _("Ruler"), _("Toggle Ruler (Ctrl+⇧+R)"));
 
-        toolbar.pack_end (settings, false, false, 0);
-        toolbar.pack_end (new Akira.Partials.Spacer (35), false, false, 0);
-        toolbar.pack_end (layout, false, false, 0);
-        toolbar.pack_end (ruler, false, false, 0);
-
-        pack_start (toolbar, true, true, 0);
+        add (menu);
+        add (new Gtk.Separator (Gtk.Orientation.VERTICAL));
```

Comment:
This spacer box thing is really dirty, there's already Gtk.Separator for this :)

Reply:
Great! I was trying to find a native solution but couldn't. I agree that the spacer is a dirty piece of garbage :D